workitem/enum_type.go

```diff
@@ -157,10 +157,17 @@ func containsAll(a []interface{}, v []interface{}) bool {
 	return result
 }
 
+// ConvertFromModel implements the FieldType interface
 func (t EnumType) ConvertFromModel(value interface{}) (interface{}, error) {
-	converted, err := t.BaseType.ConvertToModel(value)
+	if value == nil {
+		return nil, nil
+	}
+	converted, err := t.BaseType.ConvertFromModel(value)
 	if err != nil {
 		return nil, fmt.Errorf("error converting enum value: %s", err.Error())
 	}
+	if !contains(t.Values, converted) {
+		return nil, fmt.Errorf("value: %+v (%[1]T) is not part of allowed enum values: %+v", value, t.Values)
```

Comment:
@jarifibrahim please return a traceable `errs.Errorf` error instead.

Reply:
@kwk I've changed it in https://github.com/fabric8-services/fabric8-wit/pull/2224/commits/5512f135f93e5918b028ade1cb80d5f49a68814e

But could you please explain why should we use `errs.Errorf` instead of `fmt.Errorf`? Is it because `errs.Errorf` has a stacktrace?